### PR TITLE
[DEV-4520] Correcting various issues with "Award Amount" Filter

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
This started as only fixing a bug with a performance optimization for AwardAmount filter in Postgres which was discovered when comparing results to Elasticsearch. Since the fix would have left other issues unresolved on the touched code other tickets were pulled into the work.

- Frontend displaying blank cell for $0 [DEV-3736](https://federal-spending-transparency.atlassian.net/browse/DEV-3736)
- Loans with $0 obligation are returning for > $500M [DEV-3918](https://federal-spending-transparency.atlassian.net/browse/DEV-3918)
- Loan values not used in award amount filter [DEV-3315](https://federal-spending-transparency.atlassian.net/browse/DEV-3315)

**Technical details:**
- Added new derived field to some materialized views: `award_amount`
- COALESCE $ fields in materialized views
- Purged `pulled_from` from materialized views
- Updated Award Amount enum values

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [ ] Frontend
4. [x] Matview impact assessment completed
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-4520](https://federal-spending-transparency.atlassian.net/browse/DEV-4520):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected API | Script
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/awards/v2/filters/filter_helpers.py [hunk 1]
```diff
@@ -211,18 +156,16 @@ def can_use_month_aggregation(time_period):
...
-            for key, limits in WEBSITE_AWARD_BINS.items():
-                if lower_bound == limits["lower"] and upper_bound == limits["upper"]:
-                    bins.append(key)
+            for key, values in WEBSITE_AWARD_BINS.items():
+                if lower_bound == values["lower"] and upper_bound == values["upper"]:
                     break
-
-        if len(bins) == len(amount_obj):
-            return True
+            else:
```
Comment (sethstoudenmier) [inline usaspending_api/awards/v2/filters/filter_helpers.py]: Is the `else` suppose to be indented one more?

File: usaspending_api/database_scripts/matview_generator/mv_contract_award_search.json [hunk 2]
```diff
@@ -22,11 +22,12 @@
     "  awards.piid,",
     "  NULL::text AS fain,",
     "  NULL::text AS uri,",
-    "  awards.total_obligation,",
+    "  COALESCE(awards.total_obligation, 0)::NUMERIC(23, 2) AS award_amount,",
+    "  COALESCE(awards.total_obligation, 0)::NUMERIC(23, 2) AS total_obligation,",
```
Comment (sethstoudenmier) [inline usaspending_api/database_scripts/matview_generator/mv_contract_award_search.json]: Do we still need the alias `total_obligation` or is it being left to not break current functionality?

Comment (tony-sappe) [reply usaspending_api/database_scripts/matview_generator/mv_contract_award_search.json]: `total_obligation` is the value returned by the API

File: usaspending_api/awards/v2/filters/filter_helpers.py (hunk 1 above)
Comment (tony-sappe) [reply usaspending_api/awards/v2/filters/filter_helpers.py]: No, this is a `for...else`. If it was indented again to the `if` then only the first enum value would be examined

**Pull request merged**